pre_commit_hooks/check-branch-name.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+while getopts r: flag
+do
+    case "${flag}" in
+        r) BRANCH_NAME_REGEX=${OPTARG};;
+        *) echo "Invalid flag provided ($flag)"; exit 1;
+    esac
+done
+
+if [[ $BRANCH_NAME_REGEX == "" ]]; then
+  echo "No regular expression provided to validate the branch name..."
+  echo "Please provide one using the '-r' argument."
+  echo ""
+  echo "Sample 'repo' section of your '.pre-commit-config.yaml' file:"
+  echo "- repo: https://github.com/alma/pre-commit-hooks"
+  echo "  rev: v1.1.0"
+  echo "  hooks:"
+  echo "  - id: check-branch-name"
+  echo "    args:"
+  echo '      - "-r^(ci|docs|feature|fix|hotfix|perf|refactor|test)\/.*$"'
```

Comment:
```suggestion
  echo '      - "-r^(ci|docs|feature|feat|fix|hotfix|perf|refactor|test)\/.*$"'
```

Reply:
@Natim this is just an example of configuration, so it does not impact what we will actually use.

This stated, `feat` does not currently belong to officially authorized prefixes (according to [The Law](https://github.com/alma/main/blob/main/docs/contributing.md#naming-branches))... ;)

Reply:
maybe should you support the `main` branch in your example

Reply:
You mean something like:
`^(((ci|docs|feature|fix|hotfix|perf|refactor|test)\/.*)|main)$` 

Definitely can make the example more comprehensive, there is a cost on readability (of the example) though...
But really happy to add it if it is deemed useful.